### ISSUE TITLE
The new dataset Objects365 issue

### ISSUE THREAD

**Issue by runzeer** [open-mmlab/mmdetection#541]
Hi，have you tried the recent datasets Objects365? I found it similar with the COCO dataset. So I just modified the num_classes from 81 to 366, changed the image dir as well as the annotation file and the others is the same. I use the cascade rcnn config. But when I trained the model after 1 epoch ,the validation result is zero. But the loss is gradually down.  It looked abnormal. Anything else should I do?
Could you give me some advice?

**Comment by thangvubk** [comment]
To my knowledge when the number of classes is high, your score can be very low. Just try to train it more to see if the result increases or not.

**Comment by runzeer** [comment]
Thanks for your reply! But the datasets have more images than COCO.  One epoch should cost more time. So is there anything wrong with my settings?

**Comment by thangvubk** [comment]
In that case, i'm not sure. Just wait if someone can answer you.

**Comment by zimenglan-sysu-512** [comment]
hi @runzeer 
how do u download the `Objects365`?

**Comment by gxd1994** [comment]
> Hi，have you tried the recent datasets Objects365? I found it similar with the COCO dataset. So I just modified the num_classes from 81 to 366, changed the image dir as well as the annotation file and the others is the same. I use the cascade rcnn config. But when I trained the model after 1 epoch ,the validation result is zero. But the loss is gradually down. It looked abnormal. Anything else should I do?
> Could you give me some advice?

i have same problem

**Comment by nemonameless** [comment]
@ @runzeer @gxd1994  Have you train normally and converge? What is your epochs, bs, lr   setting? Thanks

**Comment by rumusan** [comment]
I think the reason is that the  order of 'categories' list in objects365_train.json an objects365_val.json is inconsistent. 

And the CocoDataset in mmdetection, the label map depends on  the  order of 'categories' list.
```  
self.cat_ids = self.coco.getCatIds(catNms=self.CLASSES) 
self.cat2label = {cat_id: i for i, cat_id in enumerate(self.cat_ids)}
```
Thus, the label in training and testing is inconsistent...

You can just modify the objects365_val.json or objects365_val.json, or the CocoDataset :
```  
self.cat_ids = self.coco.getCatIds(catNms=self.CLASSES) 
self.cat2label = {cat_id: i for i, cat_id in enumerate(sorted(self.cat_ids))}
```

**Comment by sunpeng1996** [comment]
i have same problem

**Comment by jshilong** [comment]
> I think the reason is that the order of 'categories' list in objects365_train.json an objects365_val.json is inconsistent.
> 
> And the CocoDataset in mmdetection, the label map depends on the order of 'categories' list.
> 
> ```
> self.cat_ids = self.coco.getCatIds(catNms=self.CLASSES) 
> self.cat2label = {cat_id: i for i, cat_id in enumerate(self.cat_ids)}
> ```
> 
> Thus, the label in training and testing is inconsistent...
> 
> You can just modify the objects365_val.json or objects365_val.json, or the CocoDataset :
> 
> ```
> self.cat_ids = self.coco.getCatIds(catNms=self.CLASSES) 
> self.cat2label = {cat_id: i for i, cat_id in enumerate(sorted(self.cat_ids))}
> ```

You are right.
Besides, we would support object365 and release related pre-train models in the near future

**Comment by iFighting** [comment]
> > I think the reason is that the order of 'categories' list in objects365_train.json an objects365_val.json is inconsistent.
> > And the CocoDataset in mmdetection, the label map depends on the order of 'categories' list.
> > ```
> > self.cat_ids = self.coco.getCatIds(catNms=self.CLASSES) 
> > self.cat2label = {cat_id: i for i, cat_id in enumerate(self.cat_ids)}
> > ```
> > 
> > 
> >     
> >       
> >     
> > 
> >       
> >     
> > 
> >     
> >   
> > Thus, the label in training and testing is inconsistent...
> > You can just modify the objects365_val.json or objects365_val.json, or the CocoDataset :
> > ```
> > self.cat_ids = self.coco.getCatIds(catNms=self.CLASSES) 
> > self.cat2label = {cat_id: i for i, cat_id in enumerate(sorted(self.cat_ids))}
> > ```
> 
> You are right.
> Besides, we would support object365 and release related pre-train models in the near future

will you support object365 in recent days?

**Comment by cydiachen** [comment]
> I think the reason is that the order of 'categories' list in objects365_train.json an objects365_val.json is inconsistent.
> 
> And the CocoDataset in mmdetection, the label map depends on the order of 'categories' list.
> 
> ```
> self.cat_ids = self.coco.getCatIds(catNms=self.CLASSES) 
> self.cat2label = {cat_id: i for i, cat_id in enumerate(self.cat_ids)}
> ```
> 
> Thus, the label in training and testing is inconsistent...
> 
> You can just modify the objects365_val.json or objects365_val.json, or the CocoDataset :
> 
> ```
> self.cat_ids = self.coco.getCatIds(catNms=self.CLASSES) 
> self.cat2label = {cat_id: i for i, cat_id in enumerate(sorted(self.cat_ids))}
> ```

After some tries, I observed that this modification on CocoDataset may not resolve true labels correctly. I recommend that you can directly replace the 'categories' inside val.json with 'categories' inside train.json to resolve this problem. 
For reference, I provide a sample val.json ( object365 v1 - published in 2019)

https://drive.google.com/file/d/1uDjy9FV_aOJNe8Q0bJQegAkEc4ClxjfA/view?usp=sharing
